### DISCUSSION TITLE
Does Authelia handle websockets over TLS correctly?

### DISCUSSION THREAD

**Discussion by danshat** [authelia/authelia#7834] [Q&A] [answered]
I've found a problem in Metube (self-hosted Youtube video/audio downloader project) which appears only when running behind Authelia.
Metube utilizes WSS protocol. 

I've described the issue here in detail:
https://github.com/alexta69/metube/issues/459
Here is another relevant issue:
https://github.com/alexta69/metube/issues/478

While this project is by no chance related to Authelia, perhaps someone here might have an idea of why this happens?

**Comment by james-d-elliott** [comment, marked as answer]
Provided the initialization of the websocket occurs over a secure scheme, includes the cookie, and has the appropriate headers in the authorization request authelia will respond appropriately. The protocol used at Metube is largely irrelevant to Authelia as long as it's a secure one, we treat both the [https and wss](https://github.com/authelia/authelia/blob/master/internal/utils/url.go#L42) schemes as [secure schemes](https://github.com/authelia/authelia/blob/master/internal/handlers/handler_authz.go#L31-L37). 

Based on all the communication in the links provided it seems relatively logical that the issue is Metube is not following the 302 Found responses for a number of reasons:

1. Reauthenticating with Authelia allegedly "fixes" it.
2. It was previously working and an update to Metube broke this. Yet the author has not acknowledged what changed and seems to have just ignored this factor. Have they just added websockets? Did they just become a PWA or implement other caching? Did they change websocket libraries?
